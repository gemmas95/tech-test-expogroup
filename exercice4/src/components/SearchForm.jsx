import React from "react";

function SearchForm({ handleSubmit, setRepoName }) {
  return (
    <form data-testid="form">
      <input
        placeholder="Repository name"
        onChange={(event) => setRepoName(event.target.value.trim())}
      ></input>
      <button
        data-testid="button"
        type="submit"
        onClick={(event) => {
          handleSubmit(event);
        }}
      >
        View contributors!
      </button>
    </form>
  );
}

export default SearchForm;
