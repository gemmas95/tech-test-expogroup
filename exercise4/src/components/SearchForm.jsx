import React from "react";

function SearchForm({ handleSubmit, setRepoName }) {
  return (
    <form data-testid="form" className="d-flex justify-content-center">
      <input
        placeholder="Repository name"
        onChange={(event) => setRepoName(event.target.value.trim())}
      ></input>
      <button
        className="btn btn-info"
        data-testid="button"
        type="submit"
        onClick={(event) => {
          handleSubmit(event);
        }}
      >
        View contributors!
      </button>
    </form>
  );
}

export default SearchForm;
