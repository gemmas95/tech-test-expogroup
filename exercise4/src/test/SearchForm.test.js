import renderer from "react-test-renderer";
import React from "react";
import { act } from "react-dom/test-utils";
import { render, unmountComponentAtNode } from "react-dom";
import { fireEvent } from "@testing-library/dom";

import SearchForm from "../components/SearchForm";

describe("SearchForm", () => {
  let container = null;
  const searchFormTree = renderer.create(<SearchForm />);
  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });
  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });
  it("should match snapshot", () => {
    expect(searchFormTree).toMatchSnapshot();
  });
  xit("should call onClick and go to handleSubmit", () => {
    const repoName = "heroes";
    const handleSubmit = jest.fn((event) => event.preventDefault());

    act(() => {
      render(
        <SearchForm handleSubmit={handleSubmit} setRepoName={repoName} />,
        container
      );
      const submit = document.querySelector('[data-testid="form"]');

      fireEvent.submit(submit);
    });

    const button = document.querySelector('[data-testid="button"]');

    act(() => {
      button.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    });

    expect(handleSubmit).toHaveBeenCalled();
  });
});
