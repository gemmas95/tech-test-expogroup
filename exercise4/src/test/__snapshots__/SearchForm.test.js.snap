// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match snapshot 1`] = `
<form
  className="d-flex justify-content-center"
  data-testid="form"
>
  <input
    onChange={[Function]}
    placeholder="Repository name"
  />
  <button
    className="btn btn-info"
    data-testid="button"
    onClick={[Function]}
    type="submit"
  >
    View contributors!
  </button>
</form>
`;
